download_client: Check dynamically set fragment size when downloading (#3230)
nrfconnect/sdk-nrf: doc/nrf/known_issues.rst

**Workaround:** Set :option:`CONFIG_DOWNLOAD_CLIENT_RANGE_REQUESTS`.

> NCSDK-6856: File content may be wrong when using :ref:`lib_download_client` with conten range requests.

Reviewer: ```suggestion
NCSDK-6856: File content may be wrong when using :ref:`lib_download_client` with content range requests.
```

Author: Fixed by @greg-fer

>   When using :ref:`lib_download_client` with range requests (required for TLS downloads, optionally otherwise), the HTTP response parsing may fail and cause corrupt file download.

Reviewer: ```suggestion
  When using :ref:`lib_download_client` with range requests (required for TLS downloads, optionally otherwise), the HTTP response parsing may fail and cause a corrupted file download.
```

>   Usually the download will fail before the full file content is downloaded due to failure to locate and parse HTTP header correctly, but that is not always the case.

Reviewer: ```suggestion
  Usually the download will fail before the full file content is downloaded due to failure to locate and parse HTTP headers correctly, but that is not always the case.
```
